Add version bounds to library stanza

bandali opened the pull request:
- Add some reasonable version bounds to the library stanza, leaning towards modern/newer versions of each dependency. It compiles with both GHC 8.0.1 (tested via `stack build`) and GHC 8.2.1 (tested via `cabal new-build`).
- Bump the stack resolver to latest LTS version and remove extraneous `extra-deps`.

Addresses #8.

bandali committed:
Add version bounds to library stanza

Compiles with both GHC 8.0.1 and GHC 8.2.1.

Also, bump the stack resolver to latest LTS version.

ChrisPenner commented on stack.yaml:
```diff
@@ -40,10 +40,7 @@ packages:
-- mustache-2.2.3
-- fsutils-0.1.2
-- filesystem-trees-0.1.0.6
+extra-deps: []
```
nice 👍

ChrisPenner reviewed (changes requested):
Thanks for doing this! I like it, but setting strict upper-bounds seems unnecessary; what about having a loose upper bound for minor version changes for most libs?
For instance for parsec you've listed `>= 3.1.11   && < 3.1.12`; what about just `3.1.*`?

The same for most other libs; what do you think about that?

bandali commented:
Thanks for the review. I think you have a good point. I'll push another commit to relax the upper bounds a bit, later today.

bandali committed:
Relax dependency upper bounds a bit

Also, remove superfluous dependency on `void` which I added
mistakenly in a previous PR.
